crates/matrix-sdk-indexeddb/src/event_cache_store/types.rs

```diff
@@ -94,6 +94,15 @@ impl Event {
             Event::OutOfBand(e) => e.relation(),
         }
     }
+
+    /// Sets the content of the underlying [`GenericEvent`]
+    pub fn set_content(mut self, content: TimelineEvent) -> Self {
```

Comment:
nitpick: I'd name this `with_content`, instead of `set_content`, because I'd expect `set_content` to take a `&mut self` and not return a new instance.

Reply:
fixup: [d534857b69cd3d6dba364e4db512cdd35ef3d509](https://github.com/matrix-org/matrix-rust-sdk/pull/5414/commits/d534857b69cd3d6dba364e4db512cdd35ef3d509)